pkg/model/resource/options_test.go

```diff
@@ -73,24 +74,37 @@ var _ = Describe("Resource Options", func() {
 			Expect(options.Validate().Error()).To(ContainSubstring("kind cannot be empty"))
 		})
 
-		It("should fail if the Kind is not pascal cased", func() {
-			// Base case
-			options := &Options{Group: "crew", Version: "v1", Kind: "FirstMate"}
-			Expect(options.Validate()).To(Succeed())
-
-			// Can't detect this case :(
-			options = &Options{Group: "crew", Version: "v1", Kind: "Firstmate"}
-			Expect(options.Validate()).To(Succeed())
+		DescribeTable("valid Kind values - according to core Kubernetes",
+			func(kind string) {
+				options := &Options{Group: "crew", Kind: kind, Version: "v1"}
+				Expect(options.Validate()).To(Succeed())
+			},
+			Entry("camelcase", "FirstMate"),
+			Entry("more than one caps at the start", "FIRSTMate"),
+		)
+
+		It("should fail if Kind is too long", func() {
+			var kind string
+
+			for i := 0; i < 65; i++ {
+				kind += "a"
+			}
+
+			options := &Options{Group: "crew", Kind: kind, Version: "v1"}
+			err := options.Validate()
+			Expect(err).To(MatchError(ContainSubstring("must be no more than 63 characters")))
+		})
 
-			options = &Options{Group: "crew", Version: "v1", Kind: "firstMate"}
-			Expect(options.Validate()).NotTo(Succeed())
-			Expect(options.Validate().Error()).To(ContainSubstring(
-				`kind must be PascalCase (expected FirstMate was firstMate)`))
+		It("should fail if Kind is invalid, according to core Kubernetes", func() {
+			options := &Options{Group: "crew", Kind: "Something withSpaces", Version: "v1"}
+			err := options.Validate()
+			Expect(err).To(MatchError(ContainSubstring("a DNS-1035 label must consist of lower case alphanumeric characters")))
+		})
 
-			options = &Options{Group: "crew", Version: "v1", Kind: "firstmate"}
-			Expect(options.Validate()).NotTo(Succeed())
-			Expect(options.Validate().Error()).To(ContainSubstring(
-				`kind must be PascalCase (expected Firstmate was firstmate)`))
+		It("should fail if Kind starts with a lowercase character", func() {
+			options := &Options{Group: "crew", Kind: "lOWERCASESTART", Version: "v1"}
+			err := options.Validate()
+			Expect(err).To(MatchError(ContainSubstring("Kind must start with an uppercase character")))
```

Comment:
I think you should add `Expect(err).To(HaveOccurred())` above this line to avoid panics in case the error is nil.